.github/workflows/release.yml

```diff
@@ -28,34 +28,29 @@ jobs:
           python-version: '^3.8'
 
       - name: Install Python dependencies
-        run: python -m pip install build twine
+        run: python -m pip install build scriv twine
 
       - name: Install pandoc
         run: sudo apt-get update && sudo apt-get install -y pandoc
 
       - name: Create release
         run: |
           set -x
-          new_version="$(~/auto shipit --dry-run --quiet)"
-          if [ -z "$new_version" ]
+          auto_version="$(~/auto shipit --dry-run --quiet)"
+          if [ -z "$auto_version" ]
           then echo "Nothing to release"
                exit
           fi
-          echo "[INFO] New version: $new_version"
-          changelog_version="$(perl -nle '/^# ([0-9]+\.\S*) \(.*\)/ and print $1 and exit' CHANGELOG.md)"
+          echo "[INFO] Auto version: $auto_version"
+          changelog_version="$(git tag --merged HEAD --sort v:refname | perl -nle 'print "$1.$2.", $3+1 if /^(\d+)\.(\d+)\.(\d+)$/' | tail -n1)"
           echo "[INFO] Changelog version: $changelog_version"
-          opts=
-          if dpkg --compare-versions "$new_version" lt "$changelog_version"
-          then echo "[ERROR] New version is less than latest version in changelog"
-               exit 1
-          elif dpkg --compare-versions "$new_version" eq "$changelog_version"
-          then echo '[INFO] Disabling changelog autogeneration'
-               opts=--no-changelog
-          elif ! dpkg --compare-versions "$new_version" gt "$changelog_version"
-          then echo "[ERROR] Could not compare new version '$new_version' to changelog version '$changelog_version'"
+          if [ "$auto_version" != "$changelog_version" ]
+          then echo "[ERROR] Auto version does not equal changelog version"
                exit 1
           fi
-          ~/auto shipit -vv $opts
+          scriv collect --add --version "$changelog_version"
+          git commit -m '[skip ci] Update CHANGELOG' CHANGELOG.md
+          ~/auto shipit -vv --no-changelog
```

Comment:
if `--no-changelog` -- would it trigger 
```
.autorc:                "afterChangelog": "make update-changelog && git add docs/source/changelog.rst && git commit -m '[skip ci] Update RST changelog'",
```
?  may be better to just move that call right here before `shipit`?

Reply:
Done.